code/modules/mob/dead/observer/observer.dm

```diff
@@ -283,8 +283,14 @@ Works together with spawning an observer, noted above.
 /mob/proc/ghostize(can_reenter_corpse = TRUE)
 	if(key)
 		if(key[1] != "@") // Skip aghosts.
-			if(HAS_TRAIT(src, TRAIT_CORPSELOCKED) && can_reenter_corpse) //If you can re-enter the corpse you can't leave when corpselocked
-				return
+			if(HAS_TRAIT(src, TRAIT_CORPSELOCKED))
+				if(can_reenter_corpse) //If you can re-enter the corpse you can't leave when corpselocked
+					return
+				else //otherwise, continues the ghosting process
```

Comment:
there is no reason to have the below block of code under and else case, considering the if check ends the proc with a return